linux/libio/ThrottledInput.h

```diff
@@ -0,0 +1,49 @@
+/* ============================================================
+ *
+ * Copyright (C) 2015 by Robert Haschke <rhaschke at techfak dot uni-bielefeld dot de>
+ *
+ * This file may be licensed under the terms of the
+ * GNU Lesser General Public License Version 3 (the "LGPL"),
+ * or (at your option) any later version.
+ *
+ * Software distributed under the License is distributed
+ * on an ``AS IS'' basis, WITHOUT WARRANTY OF ANY KIND, either
+ * express or implied. See the LGPL for the specific language
+ * governing rights and limitations.
+ *
+ * You should have received a copy of the LGPL along with this
+ * program. If not, go to http://www.gnu.org/licenses/lgpl.html
+ * or write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ * The development of this software was supported by:
+ *   CITEC, "Cognitive Interaction Technology" Excellence Cluster
+ *     Bielefeld University
+ *
+ * ============================================================ */
+#pragma once
+
+#include "Rate.h"
+
+namespace tactile
+{
+
+	template <typename T>
+	class ThrottledInput : public T
+	{
+	public:
+		using T::T; // inherit all constructors
+		const typename T::data_vector &readFrame() override
+		{
+			rate_.sleep();
+			return T::readFrame();
+		}
+
+		const Rate &rate() const { return rate_; }
+		void setRate(const Rate &rate) { rate_ = rate; }
+
+	private:
+		Rate rate_ = 1000.0; // default: 1kHz
```

Comment:
probably correct but my old-style programming miss the explicit operator= definition that would be clearer.

Reply:
This calls the double `constructor`.

Reply:
sure, this was the only way to explain that this works, but it is very confusing, and not "natural" for C++ where things used to be more explicit.